mypy/subtypes.py

```diff
@@ -399,6 +399,12 @@ def visit_overloaded(self, left: Overloaded) -> bool:
                     return True
             return False
         elif isinstance(right, Overloaded):
+            if left == self.right:
+                # When it is the same overload, then the types are equal.
+                # This was originally introduced as a fix / hack for:
+                # https://github.com/python/mypy/issues/9147
```

Comment:
There is no need to add a comment in the source referring to the bug (people can find that information using `git blame`).

Reply:
👍